### PR TITLE
Python vulnerability fixes

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:
Fixes the vulnerabilities found by dependabot.

Version Upgrades:
- paddlepaddle 2.5.2 - > 2.6.0
- transformers 4.35.2 -> 4.37.2
- cryptography 41.0.5 -> 42.0.2
- ~fastapi 0.95.2 -> 0.109.2~ ([starlette breaks the ray serve](https://github.com/ray-project/ray/pull/42740))
- pillow 9.5.0 -> 10.2.0
- aiohttp 3.9.0 -> 3.9.3
- ray 2.9.0 -> 2.9.2

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
**Fixes Following Vulnerabiltiies:**
**Paddle Server**
- [CVE-2024-0521](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-0521)
- [CVE-2023-52314](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52314)
- [CVE-2023-52311](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52311)
- [CVE-2023-52310](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52310)
- [CVE-2023-50447](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50447)
- [CVE-2023-50447](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50447)
- [CVE-2023-52309](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52309)
- [CVE-2023-52307](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52307)
- [CVE-2023-52304](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52304)
- [CVE-2023-52313](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52313)
- [CVE-2023-52312](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52312)
- [CVE-2023-52308](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52308)
- [CVE-2023-52306](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52306)
- [CVE-2023-52305](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52305)
- [CVE-2023-52303](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52303)
- [CVE-2023-38678](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-38678)
- [CVE-2023-38677](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-38677)
- [CVE-2023-38676](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-38676)
- [CVE-2023-38676](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-38676)
- [CVE-2023-52302](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52302)
- [CVE-2023-38674](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-38674)

**Alibi Explainer**
- [CVE-2023-6730](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-6730)
- [CVE-2023-50782](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50782)
- [CVE-2023-7018](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-7018)
- ~[CVE-2023-44271](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-44271)~
- ~[CVE-2023-5129](https://github.com/advisories/GHSA-hhrh-69hc-fgg7)~
- [CVE-2023-4863](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-4863)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)
- [CVE-2024-23334](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334)
- [CVE-2023-49083](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-49083)

**Huggingface Server**
- [CVE-2023-50782](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50782)
- [CVE-2024-23334](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)

**Kserve SDK**
- [CVE-2023-50782](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50782)
- [CVE-2024-23334](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)

**XGBoost Server**
- [CVE-2023-50782](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50782)
- [CVE-2024-23334](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)
- [CVE-2023-49083](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-49083)

**Sklearn Server**
- [CVE-2023-50782](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50782)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)
- [CVE-2024-23334](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334)
- [CVE-2023-49083](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-49083)

**Pmml Server**
- [CVE-2023-50782](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50782)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)
- [CVE-2024-23334](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334)
- [CVE-2023-49083](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-49083)

**Lgb Server**
- [CVE-2023-50782](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50782)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)
- [CVE-2024-23334](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334)
- [CVE-2023-49083](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-49083)

**Paddle Server**
- [CVE-2023-50782](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50782)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)
- [CVE-2023-49083](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-49083)

**ART Explainer**
- [CVE-2023-50447](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-50447)
- [CVE-2023-44271](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-44271)
- [CVE-2023-5129](https://github.com/advisories/GHSA-hhrh-69hc-fgg7)
- [CVE-2023-4863](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-4863)
- [CVE-2024-23829](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23829)
- [CVE-2024-23334](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334)

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Various python vulnerability fixes
```

### CODE REVIEW

File: python/alibiexplainer/pyproject.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ packages = [
 [tool.poetry.dependencies]
 python = ">=3.8,<3.12"
 kserve = { path = "../kserve", extras = ["storage"], develop = true }
-alibi = { version = "^0.9.4", extras = ["shap", "tensorflow"] }
+alibi = { version = ">=0.9.4", extras = ["shap", "tensorflow"] } # From 0.9.5 alibi uses BSL license
```
Comment (terrytangyuan) [inline python/alibiexplainer/pyproject.toml]: We cannot have alibi >= 0.9.5

Let's keep this unchanged here for now and I'll remove this as part of https://github.com/kserve/kserve/issues/3380

Comment (sivanantha321) [reply python/alibiexplainer/pyproject.toml]: My Bad. What I intended to do is <=. 😅

Comment (terrytangyuan) [reply python/alibiexplainer/pyproject.toml]: Ha that makes sense

Comment (sivanantha321) [reply python/alibiexplainer/pyproject.toml]: left it as it is.

**Pull request merged**